src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -194,19 +194,52 @@ public override IEnumerable<TypeInfo> DefinedTypes
 
         public override bool IsCollectible => GetIsCollectible(GetNativeHandle());
 
+        // GetResource will return a pointer to the resources in memory.
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        private static extern unsafe byte* GetResource(RuntimeAssembly assembly,
+                                                       string resourceName,
+                                                       out ulong length);
+
         // Load a resource based on the NameSpace of the type.
-        [System.Security.DynamicSecurityMethod] // Methods containing StackCrawlMark local var has to be marked DynamicSecurityMethod
         public override Stream GetManifestResourceStream(Type type, string name)
         {
-            StackCrawlMark stackMark = StackCrawlMark.LookForMyCaller;
-            return GetManifestResourceStream(type, name, false, ref stackMark);
+            StringBuilder sb = new StringBuilder();
+            if (type == null)
+            {
+                if (name == null)
+                    throw new ArgumentNullException(nameof(type));
+            }
+            else
+            {
```

Comment:
This can be written without the StringBuilder overhead as:

```
if (type == null && name == null)
    throw new ArgumentNullException(nameof(type));

string nameSpace = type?.Namespace;

string delimiter = (nameSpace != null && name != null) ? Type.Delimiter : null;
string resourceName = string.Concat(nameSpace, delimited, name);

return GetManifestResourceStream(resourceName);
```